File: app/src/main/java/org/quietlip/guesswhatwho/db/GameDatabase.java
package org.quietlip.guesswhatwho.db;

import android.content.Context;
import android.database.sqlite.SQLiteDatabase;
import android.database.sqlite.SQLiteOpenHelper;

import androidx.annotation.Nullable;

public class GameDatabase extends SQLiteOpenHelper {
    public static final int DATABASE_VERSION = 1;
    public static final String DATABASE_NAME = "GameScore.db";


    public GameDatabase(@Nullable Context context) {
        super(context, DATABASE_NAME, null, DATABASE_VERSION);
    }

    @Override
    public void onCreate(SQLiteDatabase db) {
        db.execSQL(DatabaseContract.SQL_CREATE_ENTRIES);
    }

    @Override
    public void onUpgrade(SQLiteDatabase db, int oldVersion, int newVersion) {
        db.execSQL(DatabaseContract.SQL_DELETE_ENTRIES);
        onCreate(db);
    }
}

File: app/src/main/java/org/quietlip/guesswhatwho/db/GameDatabase.kt
package org.quietlip.guesswhatwho.db

import android.content.Context
import androidx.room.Database
import androidx.room.Room
import androidx.room.RoomDatabase
import org.quietlip.guesswhatwho.models.Game

@Database(entities = [Game::class], version = 1)
abstract class GameDatabase : RoomDatabase(){
    abstract fun gameDao(): GameDao
    companion object {
        private var instance: GameDatabase? = null
        fun getDb(context: Context): GameDatabase?{
            if(instance == null){
                synchronized(GameDatabase::class){
                    instance = Room.databaseBuilder(context.applicationContext, GameDatabase::class.java, "game.db").build()
                }
            }
            return instance
        }
    }
}